app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -55,7 +54,6 @@ class WallpaperManager(private val settings: Settings) {
 
         if (AppCompatDelegate.getDefaultNightMode() != mode) {
             AppCompatDelegate.setDefaultNightMode(mode)
-            context.asActivity()?.recreate()
```

Comment:
Here we should only recreating the activity when the theme changed from the previous one, if it didn't change we shouldn't recreate